app/vmui/packages/vmui/src/utils/time.ts

```diff
@@ -64,12 +64,12 @@ export const formatDateForNativeInput = (date: Date): string => dayjs(date).form
 
 export const getDurationFromPeriod = (p: TimePeriod): string => {
   const dur = dayjs.duration(p.to.valueOf() - p.from.valueOf());
-  const durs: UnitTypeShort[] = ["d", "h", "m", "s"];
-  return durs
-    .map(d => ({val: dur.get(d), str: d}))
-    .filter(obj => obj.val !== 0)
-    .map(obj => `${obj.val}${obj.str}`)
-    .join(" ");
+  const ms = dur.asMilliseconds();
+  const seconds = Math.floor((ms / 1000) % 60);
+  const minutes = Math.floor((ms / 1000 / 60) % 60);
+  const hours = Math.floor((ms  / 1000 / 3600 ) % 24);
+  const days = Math.floor(ms / (1000 * 60 * 60 * 24));
+  return `${days}d ${hours}h ${minutes}m ${seconds}s`;
```

Comment:
few comments - 
1. now this will display `0d 0h 5m 0s` - not sure it is expected.
2. probably `p.to.valueOf() - p.from.valueOf()` will be amount of ms itself - so no need to create duration object itself

just curious - why this function was touched?)

Reply:
It was decided to use Chartjs when discussing in Slack ([thread](https://victoriametrics-rocks.slack.com/archives/C027VTSLBK8/p1631785325002800)).
Main pros and cons:
  1. plotlyjs (large size, configurable, based on d3)
  2. chartjs (lightweight, easy to use, configurable (covers all functionality))
  3. d3js (large size and low-level)

<hr/>

The old function ignored months and years.  
For exmaple:
  period = `{from:2021-09-01, to: 2021-10-04}`
  dayjs return `{days: 3, months: 1, ...}`
  old function returned "1d" and ingored 1 month (or 30 days)
  
  
`0d 0h 5m 0s` and `probably p.to.valueOf() - p.from.valueOf() will be amount of ms itself ` - my mistake, I will fix it

Reply:
> old function returned "1d" and ingored 1 month (or 30 days) 
good catch) makes sense

as of d3js large size - d3 is split into modules and if tree-shaking of the whole d3 does not work - the pieces can be imported from separate modules

as of configurability of chart js - I just a bit afraid that at some point Chart.js will be not enough and it will be really hard to go back to d3 :)

for instance from what I see - the legend in chart.js is also a part of canvas, so it may be hard to add some rich layout there or make the legend tabular...

also I've checked locally chartjs version and d3 version... and subjectively d3 (with rather slow material-ui html tooltips with rich layout) looks faster...

I see that the main reason to use chart.js is that "easy to use" part indeed. So if there are no resources to support and write d3 code - indeed chart.js is a way to go :)